### PR TITLE
Add support for dropping columns in Memory

### PR DESCRIPTION
**Pull request by ebyhr**
## Release notes

```markdown
# Memory
* Add support for dropping columns. ({issue}`22657`)
```

### CODE REVIEW

**Review by SemionPar — APPROVED** [review]

File: plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java [hunk 1]
```diff
@@ -388,14 +391,35 @@ public synchronized void addColumn(ConnectorSession session, ConnectorTableHandl
             throw new TrinoException(NOT_SUPPORTED, format("Unable to add NOT NULL column '%s' for non-empty table: %s", column.getName(), table.getSchemaTableName()));
         }
 
+        int maxColumnIndex = table.columns().stream()
```
Comment (nineinchnick) [inline plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java:394]: Nit:
```suggestion
        int maxColumnId = table.columns().stream()
```

File: plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryPagesStore.java [hunk 2]
```diff
@@ -117,7 +117,10 @@ public synchronized List<Page> getPages(
                 done = true;
             }
             // Append missing columns with null values. This situation happens when a new column is added without additional insert.
-            for (int j = page.getChannelCount(); j < columnIndexes.length; j++) {
+            for (int j = 0; j < columnIndexes.length; j++) {
+                if (columnIndexes[j] < page.getChannelCount()) {
```
Comment (nineinchnick) [inline plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryPagesStore.java:121]: Is this related to dropping columns, or a bug fix? Should this be in a separate commit?

File: plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java [hunk 3]
```diff
@@ -205,16 +206,18 @@ public synchronized Map<String, ColumnHandle> getColumnHandles(ConnectorSession
...
         return tables.get(handle.id())
                 .columns().stream()
-                .collect(toImmutableMap(ColumnInfo::name, ColumnInfo::handle));
+                .collect(toImmutableMap(ColumnInfo::name, c -> new MemoryColumnHandle(c.id(), c.name(), c.type(), c.nullable())));
     }
 
     @Override
     public synchronized ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
     {
-        MemoryTableHandle handle = (MemoryTableHandle) tableHandle;
```
Comment (nineinchnick) [inline plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java:214]: Why this is change necessary?

File: plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryColumnHandle.java [hunk 4]
```diff
@@ -16,7 +16,7 @@
 import io.trino.spi.connector.ColumnHandle;
 import io.trino.spi.type.Type;
 
-public record MemoryColumnHandle(int columnIndex, Type type)
+public record MemoryColumnHandle(int columnIndex, String name, Type type, boolean nullable)
```
Comment (nineinchnick) [inline plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryColumnHandle.java:19]: Are these changes required for dropping columns?
